File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> hs = new HashSet<>();
        for(int x : nums) hs.add(x);
        int ans = 0;
        for(int i=0; i<nums.length; i++) {
            if(hs.contains(nums[i]-1)) continue;
            int tmp = nums[i];
            while(hs.contains(tmp+1)) {
                tmp++;
            }
            ans = Math.max(ans, tmp-nums[i]+1);
        }
        return ans;
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-5.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        currmax, currmin = nums[0], nums[0]
        ans = nums[0]
        for i in range(1,len(nums)):
            if nums[i]==0:
                ans = max(ans, 0)
                currmax, currmin = 1,1
                continue
            tmp = currmax
            currmax = max(nums[i], currmin*nums[i], currmax*nums[i])
            currmin = min(nums[i], currmin*nums[i], tmp*nums[i])
            ans = max(ans, currmax)
        return ans